src/main/java/org/spongepowered/api/util/command/source/ProxySource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util.command.source;
+
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.service.permission.Subject;
+import org.spongepowered.api.util.command.CommandSource;
+
+/**
+ * Proxy sources are {@link CommandSource}s that are run by one source as a
+ * different entity.
+ */
+public interface ProxySource extends LocatedSource {
+
+    /**
+     * Gets the entity operated on by this proxy.
+     *
+     * @return The entity
+     */
+    Entity getEntity();
```

Comment:
@zml2008 I'm good on removing `getSubject()`, as the `Subject` methods of `ProxySource` implementations delegate to them. I'm also OK with removing `getInitiator()`, though plugins may still want to use it.